src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -857,6 +857,15 @@ public void switchFolder(FolderElement switchTo) {
 
 		upFolder.setEnabled(!currentFolder.isRoot());
 	}
+	
+	private void togglefilter(String filter) {
+		if (!Pattern.matches(".*" + filter + ".*", search.getText())) {
+			search.setText(search.getText() + filter);
+		}
+		else {
+			search.setText(search.getText().replaceAll(filter, ""));

```

Comment:
replaceAll can be replace here as you don't use regex. But consider adding another replaceAll afterwards that detects multiple spaces and merges them into one, and also add trim to the end of the statement, as otherwise filter "f:locked test" will become " test", with space we don't want.